### PR TITLE
README: Add features, description, platforms, screenshots, API.

### PR DESCRIPTION
**Pull request by shurcooL**
Add package summaries for client and server packages.

The last time I told someone (/cc @slimsag) about this project, I had to explain all the features and cool parts about it manually, because the README did a painfully bad job at it.

Once I had done that, the response was positive (much more so than after seeing the original README):

> interesting!
> okay that's really cool actually

This PR attempts to improve the situation so the project itself does a better job of explaining why it's cool.

### CODE REVIEW

File: server/main.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// Command server is the Instant Share server. It is responsible for accepting uploads and serving those files.
```
Comment (pavben) [inline server/main.go]: What's a command server? I've never heard the term.

File: client/main.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// Command client is the Instant Share client. It runs in your operating system's tray bar for quick access.
```
Comment (pavben) [inline client/main.go]: Same here. "Command" client?

File: README.md [hunk 3]
```diff
@@ -2,3 +2,39 @@ Instant Share
 =============
 
 Instantly share images, videos, and other files on the web.
+
+Features
+--------
+
+-	**Instant links**
+	-	When you share via Instant Share, you get a shareable link in your clipboard right away, so you can paste it in a conversation, in a forum post, or anywhere. That link is usable right away because Instant Share streams both the uploads and downloads in the background.
```
Comment (pavben) [inline README.md]: ... downloads in the background, allowing downloads to begin before you finish uploading. <-- maybe?

File: README.md [hunk 4]
```diff
@@ -2,3 +2,39 @@ Instant Share
...
+Instant Share server runs on OS X, Linux, Windows or any other platform that Go supports.
+
+Screenshots
+-----------
+
+![Screenshot 1](https://cloud.githubusercontent.com/assets/1924134/8891878/8dd7a024-32ee-11e5-8eca-1994f8503094.png)
+
+![Screenshot 2](https://cloud.githubusercontent.com/assets/1924134/8891877/8dc67272-32ee-11e5-9656-7f9749394ad3.png)
+
+API
```
Comment (pavben) [inline README.md]: Let's not put useless stuff in the README as it takes away from what actually matters. API is for developers, not for users. They can dig it up themselves.

File: README.md [hunk 5]
```diff
@@ -2,3 +2,39 @@ Instant Share
...
+Instant Share consists of a server and client.
+
+The client runs in your operating system's tray bar for quick access. The server is responsible for accepting uploads and serving those files.
+
+Platforms
+---------
+
+Instant Share client is fully supported on OS X. Linux and Windows clients are possible pending support being implemented in [`trayhost`](https://github.com/shurcooL/trayhost) package that Instant Share client uses for creating a tray icon.
+
+Instant Share server runs on OS X, Linux, Windows or any other platform that Go supports.
```
Comment (pavben) [inline README.md]: :+1:

File: README.md [hunk 6]
```diff
@@ -2,3 +2,39 @@ Instant Share
 =============
```
Comment (pavben) [inline README.md]: One thing to consider from a user's perspective is that the main problem this attempts to address (not knowing how to set up InstantShare) is still not addressed. Is there anything that should be added to explain how to point the client at the specific server instance and how to build the client?

File: server/main.go (hunk 1 above)
Comment (shurcooL) [inline server/main.go]: This is the standard way to document packages in Go.

If the package name is main, the package comment should begin with "// Command {{base}} ..." where base is the parent folder name (and therefore the default name of the produced binary).

If the package name is not main, the package comment should begin with "// Package {{name}} ..." where name is the package name.

Source: https://github.com/golang/go/wiki/CodeReviewComments#package-comments

Actually, it seems it only applies for non-main packages, see [here](https://github.com/golang/lint/blob/7b7f4364ff76043e6c3610281525fabc0d90f0e4/lint.go#L399-L402). So we can make this nicer. Suggestions?

File: README.md (hunk 6 above)
Comment (shurcooL) [inline README.md]: Yes, I agree, that part is not yet addressed. Let's do it in a separate PR. This PR attempts to fix the issue of describing why Instant Share is cool and useful, and what it does.

File: README.md (hunk 3 above)
Comment (shurcooL) [inline README.md]: That sounds better, I'll change it.

File: README.md (hunk 4 above)
Comment (shurcooL) [inline README.md]: Most READMEs I've seen typically have developer related stuff at the bottom. User facing stuff at the top. But we can remove it if you feel strongly about it.

File: README.md (hunk 3 above)
Comment (shurcooL) [inline README.md]: Done in 29d9fcc4fa3cb7a3779e44de0f2e940b39bbfbc6.

**Comment by shurcooL** [comment]
What's left for this to be merged?

File: README.md (hunk 4 above)
Comment (pavben) [inline README.md]: Yes, I do. It's pointless. Please remove.

File: server/main.go (hunk 1 above)
Comment (pavben) [inline server/main.go]: Eh, okay. You're more familiar than I am when it comes to Go conventions, it just seemed odd.

File: README.md (hunk 4 above)
Comment (shurcooL) [inline README.md]: Done in cae7b24ce67af8273607d437790e8806248146a8.

**Comment by shurcooL** [comment]
In a private message from @pavben:

> merge it
